src/advanced/JupyterNotebook/index.tsx

```diff
@@ -0,0 +1,322 @@
+/** *******************************************************************************************************************
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  
+      http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.                                                                              *
+ ******************************************************************************************************************** */
+
+import React, { useMemo } from 'react';
+import Grid from '../../layouts/Grid';
+import Container from '../../layouts/Container';
+import ReactMarkdown from 'react-markdown';
+import parse from 'html-react-parser';
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
+import { coy } from 'react-syntax-highlighter/dist/cjs/styles/prism';
+import { CodeComponent } from 'react-markdown/src/ast-to-react';
+import Heading from '../../components/Heading';
+import Paper from '../../layouts/Paper';
+
+// cats an array of lines together
+const sourceLines = (lines: string[]) => {
+    return lines.join('');
+};
+
+export interface JupyterNotebookProps {
+    notebookData: string;
+}
+
+interface JupyterNotebookCellProps {
+    cell: any;
+    executionCount: string;
+    language: string;
+}
+
+interface JupyterNotebookOutputCellProps extends JupyterNotebookCellProps {
+    output: any;
+}
+
+interface JupyterNotebookCellRenderTemplateProps {
+    idColumnContent: string;
+    bodyColumnContent: any;
+    bodyColumnStyle: any;
+}
+
+const JupyterNotebookCellRenderTemplate: React.FC<JupyterNotebookCellRenderTemplateProps> = ({
+    idColumnContent,
+    bodyColumnContent,
+    bodyColumnStyle,
+}) => {
+    return (
+        <>
+            <Grid item xs={1}>
+                {idColumnContent}
+            </Grid>
+            <Grid item xs={11}>
+                <Paper variant={'outlined'} style={bodyColumnStyle}>
+                    {bodyColumnContent}
+                </Paper>
+            </Grid>
+        </>
+    );
+};
+
+// -- renders a cell of type 'markdown'
+const JupyterNotebookMarkdownCell: React.FC<JupyterNotebookCellProps> = ({
+    cell,
+    executionCount: execution_count,
+    language,
+}) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={''}
+            bodyColumnContent={<ReactMarkdown children={sourceLines(cell.source)} />}
+            bodyColumnStyle={{ padding: '0px 8px 0px 8px', margin: '0px 0px', border: '1px solid #AAA' }}
```

Comment:
You just use `bodyColumnStyle` just below. So it will be better to put it to constant to reduce the code

Reply:
:+1: